google-cloud-trace/lib/google/cloud/trace/faraday_middleware.rb

```diff
@@ -0,0 +1,79 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "faraday"
+require "google/cloud/trace"
+
+module Google
+  module Cloud
+    module Trace
+      class FaradayMiddleware < Faraday::Middleware
+        ##
+        # Create a Trace span with the HTTP request/response information.
+        def call env
+          Google::Cloud::Trace.in_span "faraday_request" do |span|
+            add_request_labels span, env if span
+
+            response = @app.call env
+
+            add_response_labels span, response if span
+
+            response
+          end
+        end
+
+        protected
+
+        ##
+        # @private Set Trace span labels from request object
+        def add_request_labels span, env
+          labels = span.labels
+          label_keys = Google::Cloud::Trace::LabelKey
+
+          set_label labels, label_keys::AGENT,
```

Comment:
I suspect we don't need/want most of these. These labels should already have been set in the root (request) span. We should instead set any labels relevant to outgoing RPCs.